src/main/java/quizapp/app/favorite/FavoriteController.java

```diff
@@ -0,0 +1,64 @@
+package quizapp.app.favorite;
+
+import java.util.List;
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import quizapp.domain.model.Quiz;
+import quizapp.domain.model.User;
+import quizapp.domain.repository.quiz.QuizRepository;
+import quizapp.domain.repository.user.UserRepository;
+import quizapp.domain.service.quiz.QuizService;
+import quizapp.domain.service.user.UserDetailsImpl;
+
+@Controller
+@RequestMapping("quiz/favorites")
+public class FavoriteController {
+  @Autowired
+  QuizService quizService;
+  @Autowired
+  UserRepository userRepository;
+  @Autowired
+  QuizRepository quizRepository;
+
+  @ModelAttribute
+  FavoriteForm setUpForm() {
+    FavoriteForm form = new FavoriteForm();
+    return form;
+  }
+
+  @GetMapping
+  String favorites(Model model, @AuthenticationPrincipal UserDetailsImpl userDetailsImpl) {
+    List<Quiz> favorites = userDetailsImpl.getUser().getFavoriteQuizList();
+    model.addAttribute("favorites", favorites);
+    return "quiz/favorites";
+  }
+
+  @PostMapping
+  String favorite(@ModelAttribute FavoriteForm form, BindingResult bindingResult,
+      @AuthenticationPrincipal UserDetailsImpl userDetailsImpl, Model model) {
+    if (bindingResult.hasErrors()) {
+      return favorites(model, userDetailsImpl);
+    }
+    User currentUser = userDetailsImpl.getUser();
+    Optional<Quiz> opt = quizRepository.findById(form.getQuizId());
+    Quiz quiz = opt.get();
```

Comment:
idで検索して必ず存在するはず、という場合であれば、下記のようなコードにしていただくとBOSっぽいコードになります。

```suggestion
    Quiz quiz = quizRepository.findById(form.getQuizId()).orElseThrow();
```

Reply:
修正いたします。